src/SEnergyCorrelator.ana.h

```diff
@@ -103,12 +114,35 @@ namespace SColdQcdCorrelatorAnalysis {
       if (m_config.doPackageCalc) {
         DoLocalCalcWithPackage( m_input.jets[iJet].GetPT()  );
       }
+      if(m_config.doManualCalc) {
+	DoLocalCalcManual(m_jetCstVector, VecJet);
+      }
     }  // end jet loop
     return;
 
   }  // end 'DoLocalCalculation()'
 
-
+  double SEnergyCorrelator::GetWeight(ROOT::Math::PtEtaPhiEVector momentum, int option, optional<ROOT::Math::PtEtaPhiEVector> reference){
+    double weight = 1;
+    switch(option){
+      case Norm::Pt:{
+	weight = momentum.Pt();
+      }break;
+      case Norm::Et:{
+	TVector3 pRef(reference.value().X(), reference.value().Y(), reference.value().Z());
+	TVector3 pMom(momentum.X(), momentum.Y(), momentum.Z());
+	TVector3 pEt = pMom - (pMom*pRef/(pRef*pRef))*pRef;
+	weight = pow(pEt.Mag2()+pow(Const::MassPion(),2),0.5);
+      }break;
+      case Norm::E:{
+	weight = momentum.E();
+      }break;
+      default:{
+	weight = momentum.Pt();
+      }break;
+    }
```

Comment:
Actually, brackets aren't needed for the case statements! Also, you can use `[[fallthough]];` to indicate cases which will run the same block (e.g. `Norm::Pt` and default): 

```suggestion
    switch (option) {
      case Norm::Et:
        TVector3 pRef(reference.value().X(), reference.value().Y(), reference.value().Z());
	TVector3 pMom(momentum.X(), momentum.Y(), momentum.Z());
	TVector3 pEt = pMom - (pMom*pRef/(pRef*pRef))*pRef;
	weight = pow(pEt.Mag2()+pow(Const::MassPion(),2),0.5);
	break;
      case Norm::E:
	weight = momentum.E();
	break;
      case Norm::Pt:
        [[fallthrough]];
      default:
        weight = momentum.Pt();
        break;
    }
```